packages/terra-dropdown-button/src/SplitButton.jsx

```diff
@@ -96,13 +141,31 @@ class SplitButton extends React.Component {
   }
 
   handleCaretKeyDown(event) {
-    if (event.keyCode === KeyCode.KEY_SPACE) {
-      this.setState({ caretIsActive: true });
+    if (event.keyCode === KeyCode.KEY_SPACE || event.keyCode === KeyCode.KEY_RETURN) {
+      // In FireFox active styles don't get applied onKeyDown
+      this.setState({ caretIsActive: true, openedViaKeyboard: true });
+    } else if (event.keyCode === KeyCode.KEY_DOWN && this.state.isOpen && !this.state.openedViaKeyboard) {
+      // puts focus on first list element on down arrow key press when dropdown is opened by mouse click
+      const firstOption = this.dropdownList.querySelector('[data-terra-dropdown-first-list-item]');
+      firstOption.focus();
+      // required to prevent handleFocus() callback of DropdownList.
+      event.preventDefault();
+    } else if (event.keyCode === KeyCode.KEY_UP && this.state.isOpen && !this.state.openedViaKeyboard) {
+      // puts focus on last list element on up arrow key press when dropdown is opened by mouse click
+      const lastOption = this.dropdownList.querySelector('[data-terra-dropdown-last-list-item]');
+      lastOption.focus();
```

Comment:
Needs confirmation on keyUp. should we implement this behavior of navigating from bottom to top ? when the dropdown is opened via mouseclick.

cc. @neilpfeiffer , @PaulSolDev

Reply:
Added this functionality with reference to : https://www.w3.org/TR/wai-aria-practices-1.1/examples/menubar/menubar-2/menubar-2.html